### PR TITLE
Project: Use Dependency.visit instead of if/else

### PR DESCRIPTION
**Pull request by Geod24**
A smaller chunk of #2375 to reduce the diff (and bisect the failure).

### CODE REVIEW

File: source/dub/project.d [hunk 1]
```diff
@@ -510,17 +510,20 @@ shared static this() {
...
+							auto path = path_.absolute ? path_ : m_rootPackage.path ~ path_;
+							auto tmp = m_packageManager.getOrLoadPackage(path, NativePath.init, true);
+							return resolveSubPackage(tmp, subname, true);
+						},
+						(Repository repo) {
+							auto tmp = m_packageManager.loadSCMPackage(basename, repo);
+							return resolveSubPackage(tmp, subname, true);
+						},
+						(VersionRange range) {
+							return m_packageManager.getBestPackage(dep.name, vspec);
```
Comment (Geod24) [inline source/dub/project.d:524]: Note that until the `VersionRange` overload is introduced, this needs to use `vspec`.

Comment (WebFreak001) [reply source/dub/project.d:524]: let's add a TODO?

Comment (Geod24) [reply source/dub/project.d:524]: Well it's going to be fixed in the next PR.

**Comment by Geod24** [comment]
Self merging as it's pretty straightforward and the comment is addressed in the follow-up.

**Pull request merged**